src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1798,13 +1801,17 @@ bool Arguments::check_vm_args_consistency() {
   status = CompilerConfig::check_args_consistency(status);
 #if INCLUDE_JVMCI
   if (status && EnableJVMCI) {
+    // libjvmci doesn't require resolving jdk.internal.vm.ci as it is
+    // compiled into the libjvmci image itself. Without libjvmci, there
+    // is no other representation of the jdk.internal.vm.ci module
+    // so it needs to be added to the root module set.
+    if (ClassLoader::is_module_observable("jdk.internal.vm.ci") && !UseJVMCINativeLibrary && !_jvmci_module_added) {
```

Comment:
In which case `ClassLoader::is_module_observable("jdk.internal.vm.ci")` == `true` when `_jvmci_module_added` == `false`?
I assume this code is executed after check command line for `--add-modules=jdk.internal.vm.ci`.

Reply:
The documentation for `is_module_observable` is:
```
  // Determines if the named module is present in the
  // modules jimage file or in the exploded modules directory.
```
That is, is the module present on disk. On the other hand, `_jvmci_module_added` is a test of whether an `--add-modules` option has been seen whose value included `jdk.internal.vm.ci`.